pkg/source/control/network/mtls.go

```diff
@@ -0,0 +1,275 @@
+package network
+
+import (
+	"context"
+	"crypto/rand"
+	"crypto/rsa"
+	"crypto/tls"
+	"crypto/x509"
+	"crypto/x509/pkix"
+	"encoding/pem"
+	"errors"
+	"math/big"
+	"net"
+	"time"
+
+	"go.uber.org/zap"
+	"knative.dev/pkg/logging"
+)
+
+const (
+	organization = "knative.dev"
+	fakeDnsName  = "data-plane." + organization
+)
+
+var randReader = rand.Reader
+var serialNumberLimit = new(big.Int).Lsh(big.NewInt(1), 128)
+
+// Copy-pasted from https://github.com/knative/pkg/blob/975a1cf9e4470b26ce54d9cc628dbd50716b6b95/webhook/certificates/resources/certs.go
+func createCertTemplate() (*x509.Certificate, error) {
```

Comment:
plan to refactor the reference for reuse?

Reply:
I'm working here https://github.com/slinkydeveloper/control-data-plane-communication/tree/certificates to develop a kind of "certificates reconciler" that will take care of certs generation both for control plane and data plane. I'll get back soon with more details